App/TabTemplate.cs

```diff
@@ -155,32 +145,36 @@ private void listView1_SelectedIndexChanged(object sender, EventArgs e)
 
         }
 
-        private void buttonDirectory_Click(object sender, EventArgs e)
-        {
-            using (var folderDialog = new FolderBrowserDialog())
-            {
-                folderDialog.SelectedPath = _appPath;
-                DialogResult result = folderDialog.ShowDialog();
-
-                if (result == DialogResult.OK && !string.IsNullOrWhiteSpace(folderDialog.SelectedPath))
-                {
-                    string[] files = Directory.GetFiles(folderDialog.SelectedPath);
-
-                    _fileListAuthorize.Add(Path.GetFileName(folderDialog.SelectedPath), folderDialog.SelectedPath);
-                    addDirList(folderDialog.SelectedPath);
-                }
-            }
-        }
-
         private void buttonFile_Click(object sender, EventArgs e)
         {
             System.Windows.Forms.OpenFileDialog fileDialog = new System.Windows.Forms.OpenFileDialog();
+            fileDialog.ValidateNames = false;
             fileDialog.InitialDirectory = _appPath;
+            fileDialog.Multiselect = true;
+            fileDialog.CheckPathExists = true;
+            fileDialog.CheckFileExists = false;
+            fileDialog.FileName = "Default";
 
             if (fileDialog.ShowDialog() == DialogResult.OK)
             {
-                addDirList(fileDialog.FileName);
-                _fileListAuthorize.Add(Path.GetFileName(fileDialog.FileName), fileDialog.FileName);
+                if (File.Exists(fileDialog.FileName))
+                {
+                    addDirList(fileDialog.FileName);

```

Comment:
```suggestion
                    addPathToList(fileDialog.FileName);

```
I think this method should be rename since it doesn't add only directory to  a list

Reply:
Done

Reply:
Done